app/src/main/java/com/flatstack/android/DI.java

```diff
@@ -0,0 +1,69 @@
+package com.flatstack.android;
+
+import android.content.Context;
+import android.os.Environment;
+
+import com.jakewharton.picasso.OkHttp3Downloader;
+import com.squareup.picasso.Picasso;
+
+import java.io.File;
+
+import okhttp3.OkHttpClient;
+
+/**
+ * Created by Ilya Eremin on 14.03.2016.
+ */
+
+/**
+ * When you need different dependencies for tests and prod
+ * then create new flavors: mock and producation and move these dependencies there
```

Comment:
just a thought:
flavors for mock singletons will turn out a greater mess than the Dagger mess
but still, I'm not sure if we can do better here